<launch>
  <!-- publish tf of respeaker -->
  <arg name="publish_tf" default="true"/>
  <!-- launch sound_play node -->
  <arg name="launch_soundplay" default="true"/>
  <!-- langage used in speech_to_text.py -->
  <arg name="language" default="en-US"/>
  <!-- self_cancellation means halting speech_to_text while the robot is playing sound -->
  <arg name="self_cancellation" default="true"/>

  <node if="$(arg publish_tf)" name="static_transformer" pkg="tf" type="static_transform_publisher"
        args="0 0 0 0 0 0 map respeaker_base 100"/>

  <node name="respeaker_node" pkg="respeaker_ros" type="respeaker_node.py"
	output="screen"/>

  <node if="$(arg launch_soundplay)"
        name="sound_play" pkg="sound_play" type="soundplay_node.py"/>

  <node name="speech_to_text" pkg="respeaker_ros" type="speech_to_text.py">
    <rosparam>
      language: $(arg language)
      self_cancellation: $(arg self_cancellation)
      tts_tolerance: 0.5
    </rosparam>
  </node>
</launch>
